game/src/main/scala/io/aigar/game/Virus.scala

```diff
@@ -10,43 +10,48 @@ object Virus {
   final val RespawnRetryAttempts = 15
 }
 
-class Virus(var position: Vector2) {
-  def update(grid: Grid, players: List[Player]): Unit = {
-    val cell = detectCollisions(players)
-     cell match {
-      // TODO Add the virus consumption into the cell
-      case Some(c: Cell) => respawn(grid, players)
-      case _ =>
-    }
+class Virus(spawnPosition: Vector2) extends Entity {
+  position = spawnPosition
+  mass = Virus.Mass
+
+  def state: Position = {
+    position.state
   }
+}
 
-  def detectCollisions(players: List[Player]): Option[Cell] ={
-    for(player <- players) {
-      for(cell <- player.cells) {
-        // TODO Change the 1.1 value to the constant
-        if(cell.contains(position) && cell.mass > Virus.Mass * Cell.MassDominanceRatio){
-          return Some(cell)
-        }
+class Viruses(grid: Grid) extends EntityContainer {
+  var viruses = List.fill(Virus.Quantity)(new Virus(grid.randomPosition))
+
+  def update(grid: Grid, players: List[Player]): Unit = {
```

Comment:
I'm pretty sure the `update` could be generic and managed by `EntityContainer`. And the container could have methods like `def shouldRespawn(): Boolean` which would be like that for viruses 
```scala
def shouldRespawn(): Boolean = viruses.size < Virus.Quantity
```

Could the `detectCollisions` be generic too?